### PR TITLE
Added Dockerfile and updated README

### PR DESCRIPTION
**Pull request by jonbullock**
Resolves jbake-org/jbake-docker-image#2 jbake-org/jbake-docker-image#3

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+USER ${JBAKE_USER}
+
+COPY --from=builder /opt/jbake /opt/jbake
+
+WORKDIR /mnt/site
+
+VOLUME ["/mnt/site"]
+
+ENTRYPOINT ["jbake"]
+CMD ["-b","-s"]
```
Comment (manikmagar) [inline Dockerfile:34]: Should we put `-s` in the command? Most of the use case could be run this as a part of build pipeline. 
To just bake the site without serving `-b` argument will need to be provided to override default CMD - 

```
docker run --rm -u jbake -v "$PWD":/mnt/site jbake/jbake:latest -b
```

Other option is to default it to "-b" but then users who will use it to server locally, will require to override CMD with `-b -s`.

Eitherway, I think we should document this in README.

Comment (ancho) [reply Dockerfile:34]: Yes. That's a good idea.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,36 @@
+FROM openjdk:8-alpine as builder
```
Comment (ancho) [inline Dockerfile:1]: We should switch to a more docker friendly version of the jdk/jre images.
Smaller footprint. Better jvm memory consumption.
I started playing around with the latest version from adoptopenjdk which works very well.
See https://github.com/ancho/jbake/blob/feature/docker-multistage-build/Dockerfile

**Review by ancho — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (jonbullock) [reply Dockerfile:34]: Yeah good point, I'll default to just `-b` and update the README.

File: Dockerfile (hunk 2 above)
Comment (jonbullock) [reply Dockerfile:1]: Yeah happy to go with adoptopenjdk, I'll update now.

Comment (jonbullock) [reply Dockerfile]: Had to revert to adoptopenjdk/openjdk11 as Gradle 6.8.3 didn't like openjdk16, but noticed your branch had an update to Gradle 7 so we can upgrade to openjdk16 at the same time.

File: README.asciidoc [hunk 3]
```diff
@@ -54,6 +54,41 @@ Talk to other users of JBake on the forum:
 
 * http://groups.google.com/group/jbake-user[Forum]
 
+== Docker Image
+
+The image uses the official openjdk:8-alpine image for building a distribution of JBake and openjdk:8-jre-alpine for runtime.
```
Comment (manikmagar) [inline README.asciidoc:59]: This doesn't look correct anymore. New change is using JDK11 to build and JRE11 to run.

**Review by manikmagar — CHANGES_REQUESTED** [review]

File: README.asciidoc (hunk 3 above)
Comment (jonbullock) [reply README.asciidoc]: Ah yeah thanks for catching that.

**Review by manikmagar — APPROVED** [review]
LGTM

**Pull request merged**